src/actions/app.js

```diff
@@ -34,6 +34,25 @@ export const { getDatasetById } = createActions({
   [ActionTypes.GET_DATASET_BY_ID_SUCCESS]: dataset => dataset,
 });
 
+export const { getDatasetPolicy } = createActions({
+  [ActionTypes.GET_DATASET_POLICY_SUCCESS]: policy => policy,
+  [ActionTypes.GET_DATASET_POLICY]: policy => policy,
+});
+
+export const { setDatasetPolicy } = createActions({
+  [ActionTypes.SET_DATASET_POLICY]: (datasetId, users) => ({
```

Comment:
is this just for readers? since you have a remove readers, maybe this should be ADD_READER_TO_DATASET?